components/switch/src/switch.js

```diff
@@ -0,0 +1,165 @@
+/** @jsx jsx */
+
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { jsx, useTheme, paint } from '@westpac/core';
+
+import { SwitchText, SwitchToggle } from './styled';
+
+// ==============================
+// Utils
+// ==============================
+
+// ==============================
+// Component
+// ==============================
+
+export const Switch = ({
+	name,
+	id,
+	toggleText,
+	size,
+	block,
+	flip,
+	srOnlyText,
+	defaultChecked,
+	disabled,
+	children,
+	...props
+}) => {
+	const { switch: formSwitch } = useTheme();
+
+	const [isChecked, setIsChecked] = useState(defaultChecked);
+	const toggle = () => setIsChecked(!isChecked);
+
```

Comment:
Since we want this to act as a controlled component we should be mapping the current checked state to the app state i.e. should be dependent on the checked prop passed in. Need to add useEffect hook to update our state when new props are passed in
```
useEffect(
		() => {
			setIsChecked(checked);
		},
		[checked]
	);
```